Scripts/Runtime/Job/AccessControl/AccessControlledValue.cs

```diff
@@ -28,6 +28,7 @@ public AccessControlledValue(T value)
 
         protected override void DisposeSelf()
         {
+            m_AccessController.Acquire(AccessType.Disposal);
             m_AccessController.Dispose();
             (m_Value as IDisposable)?.Dispose();
             base.DisposeSelf();
```

Comment:
If `m_Value` isn't disposable should we really require a main thread sync on the write handle?

```suggestion
            if (m_Value is IDisposable disposable)
            {
                m_AccessController.Acquire(AccessType.Disposal);
                disposable.Dispose();
            }
            m_AccessController.Dispose();
            
            base.DisposeSelf();
```

Reply:
Another thought. There is a comment in `AccessController` to call complete on the relevant job handles when we call `AccessController.Dispose` should we just do that instead? That seems like a more robust solution.

Reply:
Right, ok that makes sense. Sounds like as a future feature we may want to make a burstable version of AccessControlledValue and/or some pattern to set off a main thread task after a job handle is complete.

> When we dispose an AccessController, we're just saying that we can no longer use that AccessController, but the already scheduled jobs that are in flight I think are fine to continue onward.

This feels uncomfortable but makes sense. I guess we'll find out if this comes back to bite us one day.